pages/blog/typed-document-node.mdx

```diff
@@ -0,0 +1,106 @@
+
+<Gfycat gifId={'pastfatalirishwaterspaniel'} />
+
+### 2020: _NEW_ `TypedDocumentNode`
+
+Generating code is nice, but we don't always need to wrap hooks, services or similar code with more code. With the power of TypeScript, we can pre-compile the GraphQL operation into a `DocumentNode`, and add burn-in the TypeScript types.
+
+With the support of the client-side libraries, we get automatic type inference and auto-complete - without generating additional code:
+
+<Gfycat gifId={'joyfulableasianwaterbuffalo'} />
+
+## Live Demo
+
+[You can try it live here](https://codesandbox.io/s/github/dotansimha/graphql-typed-document-node/tree/master/examples/apollo-client-3), note the autocomplete and automatic type inference for `result` variable.
+
+## How it works?
+
+This project works in the following way:
+
+1. You write your GraphQL operations (`query` / `mutation` / `subscription` / `fragment`) in any way your prefer (for example - in a `.graphql` file).
+2. [GraphQL Code Generator](https://graphql-code-generator.com/) will generate a `TypedDocumentNode` for your operations (which is a bundle of pre-compiled `DocumentNode` with the operation result type and variables type).
+3. Instead of using your `.graphql` file - import the generated `TypedDocumentNode` and use it with your GraphQL client framework.
+4. You'll get automatic type inference, auto-complete and type checking based on your GraphQL operation.
+
+The definition of `TypedDocumentNode` is super simple - it's all about the TS generics:
+
+```ts
+import { DocumentNode } from 'graphql';
+export interface TypedDocumentNode<Result = {}, Variables = {}>
+  extends DocumentNode {}
+```
+
+### Can I use it now?
+
+This library is already available to use, but it requires you to setup it in your project, since `TypedDocumentNode` isn't supported automatically in all GraphQL client libraries.
+
+That's why we use TypeScript powers to overload existing declarations, and added support to other libraries without effecting it's runtime.
+
+[You can find a list of all supported GraphQL clients here](https://github.com/dotansimha/graphql-typed-document-node#supported-libraries) and [a short getting started tutorial here](https://github.com/dotansimha/graphql-typed-document-node#how-to-use).
+
+If you are using a library that isn't supported yet, [you can always add support to it manually using method overloading](https://github.com/dotansimha/graphql-typed-document-node#how-to-extend-other-libraries-with-this).
+
+## What's next?
+
+GraphQL client libraries [can easily add support](https://github.com/dotansimha/graphql-typed-document-node#how-can-i-support-this-in-my-library) for `TypedDocumentNode`, without breaking any API, allowing developers to have direct support for that, without the need for the supporing libraries.
+
+And, maybe, one day, it will part of the [original GraphQL `DocumentNode` interface](https://github.com/graphql/graphql-js/blob/master/src/language/ast.d.ts#L212-L216) ;)
```

Comment:
```suggestion
And, maybe, one day, it will part of the [original GraphQL `DocumentNode` interface](https://github.com/graphql/graphql-js/blob/4b7ab47c3f4ea6be99c3d9754b21fe626f7c7177/src/language/ast.d.ts#L212-L216) ;)
```